src/Avalonia.Input/InputElement.cs

```diff
@@ -540,5 +533,121 @@ private void UpdateIsEnabledCore(InputElement parent)
                 child.UpdateIsEnabledCore(this);
             }
         }
+
+        #region OnTextInput compatibility layer
+
+        private static Dictionary<Type, bool> s_compatibleTextInputRegistry = new Dictionary<Type, bool>();
+        private CompatibleTextInputHandler _compatibleTextInputHandler;
+        class CompatibleTextInputHandler : ITextInputHandler
```

Comment:
If this is any more than WIP code it should be moved to a separate class rather than putting it in a `#region`.